File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int n = nums.length;
        int[] left = new int[n];
        Arrays.fill(left, Integer.MAX_VALUE);
        int[] right = new int[n];
        Arrays.fill(right, Integer.MIN_VALUE);
        
        int smallest_on_left = Integer.MAX_VALUE;
        for(int i = 0; i < n; i++) {
            left[i] = smallest_on_left;
            smallest_on_left = Math.min(smallest_on_left, nums[i]);
        }
        
        int largest_on_right = Integer.MIN_VALUE;
        for(int i = n-1; i >=0; i--) {
            right[i] = largest_on_right;
            largest_on_right = Math.max(largest_on_right, nums[i]);
        }
        
        for(int i = 0; i < n; i++) {
            if(left[i] < nums[i] && nums[i] < right[i]) {
                return true;
            }
        }
        return false;
    }
}

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    
    let left = []
    let right = []
    
    let smallest_on_left = Infinity
    for(let i = 0; i < nums.length; i++) {
        left[i] = smallest_on_left
        smallest_on_left = Math.min(smallest_on_left, nums[i])
    }
    let largest_on_right = -Infinity
    for(let i = nums.length - 1; i >=0; i--) {
        right[i] = largest_on_right
        largest_on_right = Math.max(largest_on_right, nums[i])
    }
    //console.log(left, right)
    for(let i = 0; i < nums.length; i++) {
        if(left[i] < nums[i] && nums[i] < right[i]) {
            return true
        }
    }
    return false
};